.github/workflows/main.yml

```diff
@@ -103,6 +106,12 @@ jobs:
           cd build
           cpack
 
+      - name: Test
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+          git clone https://github.com/rfomin/quickcheck.git
```

Comment:
I'd prefer to have the testing infrastructure in the woof repo. We don't have to add the PWADs and demo lumps as files, download links (and `.gitignore` entries) should be sufficient.

Reply:
But where do we download the files from? From the DSDA website?
We could implement a submodule like in Choco.

`make` is hard to use in native Windows environment, maybe we should rewrite quickcheck in Python.

Reply:
> But where are we going to download the files from? From the DSDA website? 

Yep, we are doing the same in Debian's game-data-packager tool:

https://salsa.debian.org/games-team/game-data-packager/-/blob/master/data/compet-n.yaml

> `make` is hard to use in native Windows environment

Not in MSYS2? 😉

Reply:
> Yep, we are doing the same in Debian's game-data-packager tool

Why not the idgames archive? Also what is the purpose of this tool? These PWADs are certainly classics, but other than scythe.wad and av.wad they honestly haven't aged well.

Anyway, I think GitHub hosting is more reliable for our purpose.
 
> Not in MSYS2? 😉

Yes, it's even annoying to setup in `win_msvc.yml`. Makes me want to rewrite it.

Reply:
> Why not the idgames archive? 

The DSDA archive contains the exact versions of the WADs as used by speed runners. Also, it often contains them in a convenient form, when they are sometimes wrapped into "installers" or need some esoteric dehackery or nwtmerge stuff.

> Also what is the purpose of this tool? 

It's used to package proprietary game assets (and semi-official addons, which is where the Compet-N maps come from) for open-source game-engines in Debian's DEB package format.

> These PWADs are certainly classics, but other than scythe.wad and av.wad they honestly haven't aged well.

But they form a publically acclaimed set of PWADs which were considered on-par with the IWADs - at least back then. They were the few slected PWADs that the Compet-N community would accept speed runs for. Things sure have changed in the age of DSDA, but these were the rules back then.

Reply:
> But they form a publically acclaimed set of PWADs which were considered on-par with the IWADs - at least back then.

This may be a controversial opinion, but barring Doom 1 episode 1 and a few maps, IWADs are pretty bad by today's standards 😄 How about Cacowards?

Reply:
So, if we want to download wads/demos, I'll just write a script in Python. Should I give up on the quickcheck idea?

Reply:
I'd say we can have both, e.g. download ZIP files with `wget` and add rules to the Makefile to do this automatically if the files aren't available, yet.

Reply:
Of course, we can further modify quickcheck, but at this point we can completely rewrite it. Python works much better on Windows

Reply:
That'd be fine with me. 😉